types/xdg_shell/wlr_xdg_toplevel.c

```diff
@@ -11,50 +11,67 @@ void handle_xdg_toplevel_ack_configure(
 		struct wlr_xdg_surface *surface,
 		struct wlr_xdg_surface_configure *configure) {
 	assert(surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL);
-	assert(configure->toplevel_state != NULL);
 
-	surface->toplevel->last_acked = *configure->toplevel_state;
+	struct wlr_xdg_toplevel_configure *acked = configure->toplevel_configure;
+	assert(acked != NULL);
+
+	surface->toplevel->pending.maximized = acked->maximized;
+	surface->toplevel->pending.fullscreen = acked->fullscreen;
+	surface->toplevel->pending.resizing = acked->resizing;
+	surface->toplevel->pending.activated = acked->activated;
+	surface->toplevel->pending.tiled = acked->tiled;
+
+	surface->toplevel->pending.width = acked->width;
+	surface->toplevel->pending.height = acked->height;
 }
 
-bool compare_xdg_surface_toplevel_state(struct wlr_xdg_toplevel *state) {
+bool compare_xdg_surface_toplevel_state(struct wlr_xdg_toplevel *toplevel) {
 	// is pending state different from current state?
```

Comment:
This comment is not accurate, this function checks if the scheduled state is different from the pending state. Would you mind correcting it since you're already touching this function?

Reply:
Replaced with a comment about comparing the scheduled configure to the last sent one.

Reply:
Actually the comment was incorrect, as the scheduled configure was compared to the last *acked* one, but it doesn't matter as the whole (badly named btw) function is removed now.